### DISCUSSION TITLE
Vercel tokens disappeared - Github Actions throws "Error! You do not have access to the specified account" but it was working fine a few weeks ago

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/vercel#5098] [Help] [answered]
I used to be able to automatically deploy a Vercel deployment through Github Actions. I use a `ZEIT_TOKEN ` github secret to store my vercel API token. 

It was working fine since April, and I noticed today that it now fails with:

```
Vercel CLI 19.2.0
19.2.0
npx: installed 104 in 13.977s
npx: installed 104 in 3.438s
> [debug] [2020-08-25T17:46:12.894Z] Found config in provided --local-config path /home/runner/work/next-right-now/next-right-now/now.customer1.staging.json
Vercel CLI 19.2.0
> [debug] [2020-08-25T17:46:12.907Z] user supplied a possible target for deployment
Error! You do not have access to the specified account
Learn More: https://err.sh/now/scope-not-accessible
error Command failed with exit code 1.
##[error]Process completed with exit code 1.
```

You can see the full logs at:
https://github.com/UnlyEd/next-right-now/runs/1027842857

I'm wondering what might have caused that. I'm still part of the team, the `scope` hasn't changed, the `ZEIT_TOKEN` neither. Can it expire? 

I encountered a similar issue locally, but it was fixed by running `now login`. But I can't do that on CI/CD. 

Also, it might be a coincidence, but it's happening today after 2 weeks of holidays, so maybe the token got invalidated for not being used for a little while?

While writing this, I went to https://vercel.com/account/tokens and I noticed that all my tokens have disappeared. I had created a few tokens dedicated to Github Actions and they're gone.

![image](https://user-images.githubusercontent.com/3807458/91213920-9e54b300-e712-11ea-93d1-5c2544717868.png)

So, I suppose that's why it doesn't work anymore, but I don't explain why those tokens would disappear on their own.

**Comment by Vadorequest** [comment]
I can confirm generating a new Vercel token fixed my Github Actions workflow. (deployment failed, but for another reason)

https://github.com/UnlyEd/next-right-now/runs/1028056727

**Reply by willemliu** [reply]
I had the same issue. The timing coincided with my personal account being disabled because of over-usage. So I thought my tokens also expired because of that status change. The tokens were still unusable even after re-activation of my account. I had to regenerate the tokens.

**Reply by Vadorequest** [reply]
@Timer This is a bit concerning and you might want to take a look at it.

Maybe this is expected on your side, but certainly not on mine, all my CI/CD actions basically broke on their own due to this. I'm glad I hadn't configured too many secrets, because remembering which one I had set up and where would certainly be a challenge 😨

**Comment by styfle** [comment, marked as answer]
> it's happening today after 2 weeks of holidays, so maybe the token got invalidated for not being used for a little while?

Indeed, that is what happened 👍 

The token's time to expiration has been decreased to 10 days https://vercel.com/blog/changelog-august-2020

cc @javivelasco

**Reply by Vadorequest** [reply]
Is it configurable somehow? Or is it possible (at least) to be notified of such automated invalidation?

As I use those tokens for CI/CD with Github Actions, it doesn't make sense for them to be invalidated for such a short period of inactivity. And if it ever does, it'd be much easier to fix it if I'm aware of it.

**Reply by styfle** [reply]
It's not configurable. If you're using custom CI/CD, then you can run a cron job to periodically refresh the token.

Otherwise, you can use the official [Vercel for GitHub Integration](https://vercel.com/github) and avoid tokens altogether.

**Reply by Vadorequest** [reply]
The official Vercel for Github integration isn't suited for our use case (we didn't build a custom GHA pipeline just for fun 😉).

That's unfortunate it's not configurable, is it planned to add a notification system though? That'd ease the pain I've gone through quite a bit.

**Reply by willemliu** [reply]
> Tokens will now expire after 10 days of inactivity.

This is apparently not true. Our token expired again today however we use the token daily (except weekends). A notification of expiration of the token would be very much appreciated.
The last time the token expired was 16-aug and it expired again today (27-aug). So it seems to expire every 10 days regardless of activity.

**Reply by styfle** [reply]
> we use the token daily

How is the token used? Creating deployments with Vercel CLI?

Try visiting https://vercel.com/account/tokens and look at the "Last Active" column to see when the token was last used. Then you can make a deployment with Vercel CLI and confirm that "Last Active" changes to 0s.

**Reply by Vadorequest** [reply]
Also, I suggest adding some information to https://vercel.com/account/tokens page to explicit this behaviour of 10 days before deletion.

**Reply by willemliu** [reply]
> > we use the token daily
> 
> How is the token used? Creating deployments with Vercel CLI?
> 
> Try visiting https://vercel.com/account/tokens and look at the "Last Active" column to see when the token was last used. Then you can make a deployment with Vercel CLI and confirm that "Last Active" changes to 0s.

It's used in a custom CI/CD (just like @Vadorequest). That column says it's last active 2d ago.

**Reply by Vadorequest** [reply]
I just tested it, and my own GH Actions properly triggered a token "last active" reset, as expected. (a push on the repo started a Vercel deployment and used the token to do so)

**Reply by styfle** [reply]
> That column says it's last active 2d ago.

Is that accurate? Did you last use it two days ago or did you use it today?

**Reply by willemliu** [reply]
> > That column says it's last active 2d ago.
> 
> Is that accurate? Did you last use it two days ago or did you use it today?

It's accurate and because it's been invalidated today new deployments didn't work and the token no longer refreshes the active date.

Screenshot from pipeline:
![image](https://user-images.githubusercontent.com/974906/91453724-679cab00-e880-11ea-8bcc-5b289652b035.png)

Last Active:
![image](https://user-images.githubusercontent.com/974906/91453777-75523080-e880-11ea-9d35-14c8c9a93050.png)

**Reply by styfle** [reply]
This shows green and red but it doesn't show which commands are executed. Can you provide more information?

Consider making a support request if you don't want to share publicly https://vercel.com/support/request

**Comment by Vadorequest** [comment]
Follow-up in https://github.com/vercel/vercel/discussions/5112. I opened a suggestion to improve DX regarding this issue. 

Auto-removal of tokens is the wanted behavior for Vercel, but comes as unexpected from a developer standpoint.
